packages/tools/src/tools/annotation/planarFreehandROITool/openContourEditLoop.ts

```diff
@@ -0,0 +1,548 @@
+import { getEnabledElement } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { state } from '../../../store';
+import { Events } from '../../../enums';
+import {
+  resetElementCursor,
+  hideElementCursor,
+} from '../../../cursors/elementCursor';
+import type { EventTypes, Annotation } from '../../../types';
+import { vec3, vec2 } from 'gl-matrix';
+import { polyline } from '../../../utilities/math';
+import triggerAnnotationRenderForViewportIds from '../../../utilities/triggerAnnotationRenderForViewportIds';
+
+const { addCanvasPointsToArray, getSpacingAndXYDirections } = polyline;
+
+/**
+ * Activates the open contour edit event loop.
+ */
+function activateOpenContourEdit(
+  evt: EventTypes.MouseDownActivateEventType,
+  annotation: Annotation,
+  viewportIdsToRender: string[]
+): void {
+  this.isEditingOpen = true;
+
+  const eventDetail = evt.detail;
+  const { currentPoints, element } = eventDetail;
+  const canvasPos = currentPoints.canvas;
+  const enabledElement = getEnabledElement(element);
+  const { viewport } = enabledElement;
+
+  const prevCanvasPoints = annotation.data.polyline.map(viewport.worldToCanvas);
+
+  const { spacing, xDir, yDir } = getSpacingAndXYDirections(
```

Comment:
this is not spacing as I understand, it is spacing/subPixelResolution, can we name it differently so that it does not get confused with image spacings?

Reply:
Yeah, its derived from image spacing, but is a different concept. Renamed.